macOS/DuckDuckGo/Tab/Model/Tab+UIDelegate.swift

```diff
@@ -244,6 +244,32 @@ extension Tab: WKUIDelegate, PrintingUserScriptDelegate {
         }
     }
 
+    @objc(_webView:requestStorageAccessPanelForDomain:underCurrentDomain:completionHandler:)
+    @available(macOS 10.14, iOS 12.0, *)
+    func webView(_ webView: WKWebView,
+                 requestStorageAccessPanelForDomain requestingDomain: String,
+                 underCurrentDomain currentDomain: String,
+                 completionHandler: @escaping (Bool) -> Void) {
+        let alert = NSAlert.storageAccessAlert(currentDomain: currentDomain,
```

Comment:
@tomasstrba should we consider using `JSAlertController` instead so the alert doesn‘t block the window and allows quick opt-out of the "scary" alert?

Reply:
Yeah, it's a good suggestion. 👍 There are a couple of suggestions in the Ship Review to make the view more structured, so I believe it's a good combination.

Reply:
Unfortunately, the dialog has to be blocking since the completion handler is expected to be called during the execution of the method. Otherwise, there is an exception